### ISSUE TITLE
Proposal: `no-common-default-case`

### ISSUE THREAD

**Issue by Yash-Singh1** [eslint/eslint#14924]
**Please describe what the rule should do:**

Let's say I have the following code:

```js
switch (num) {
    case 1:
		// code

	case 2:
	default:
		// default stuff
}
```

There would be no reason to have a common case with a `default` case because anyways, it will fall back to the `default` case. So the above should fix to:

```js
switch (num) {
	case 1:
		// code

	default:
		// default stuff
}
```

**What new ECMAScript feature does this rule relate to?**

None

**What is the category of this rule? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Other (please specify: Removes useless switch-cases)

**Provide 2-3 code examples that this rule will warn about:**

```js
switch (num) {
    case 1:
		// code

	case 2:
	default:
		// default stuff
}

switch (num) {
    case 1:
		// code

	case 4:
		// stuff

	case 2:
	case 3:
	default:
		// default stuff
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

ESLint usually has rules that handle use cases that are broader or apply to all projects. Plugins are for projects that use a specific framework, for example, React.

**Are you willing to submit a pull request to implement this rule?**

Sure

**Comment by nzakas** [comment]
Thanks for the submission. Unfortunately, this rule doesn't meet our [criteria for new rules](https://eslint.org/blog/2020/05/changes-to-rules-policies), so I'm closing this pull request. Please note that you can always create any rule you want and distribute it in your own plugin for use with ESLint.
